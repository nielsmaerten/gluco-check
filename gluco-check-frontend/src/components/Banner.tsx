import { useTranslation } from "react-i18next";
import {
  Box,
  Button,
  Container,
  Divider,
  Fab,
  Grid,
  makeStyles,
  Paper,
  Typography,
} from "@material-ui/core";
import { Warning } from "@material-ui/icons";
import { JOIN_BETA } from "../lib/constants";

const useStyles = makeStyles((theme) => ({
  title: {
    fontSize: theme.typography.body1.fontSize,
    fontWeight: 600,
    color: theme.palette.text.secondary,
  },
  subtitle: {
    fontSize: theme.typography.body2.fontSize,
  },
  button: {
    fontWeight: 600,
    backgroundColor: theme.palette.warning.main,
    color: theme.palette.warning.contrastText,
  },
}));

function Banner() {
  const { t, i18n } = useTranslation();

  // TODO: can't we just get this straight from i18n?
  const currentLngName = t(
    `languageSelector.availableLanguageLabels.${i18n.language}`
  );

  const classes = useStyles();

  function joinBeta() {
    window.location.assign(`${JOIN_BETA}/${i18n.language}`);
  }

  return (
    <>
      <Paper elevation={1}>
        <Container maxWidth="md">
          <Box pt={2} pr={1} pb={1} pl={2}>
            <Grid container spacing={2} alignItems="flex-start" wrap="nowrap">
              <Grid item>
                <Fab component="div">
                  <Warning />
                </Fab>
              </Grid>
              <Grid item>
                <Typography className={classes.title} color="secondary">
                  {t("settings.betaBanner.title")}
                </Typography>
                <Typography className={classes.subtitle}>
                  {t("settings.betaBanner.message", {
                    language: currentLngName,
                  })}
                </Typography>
              </Grid>
            </Grid>
            <Grid container justify="flex-end" spacing={8}>
              <Grid item>
                <Button
                  className={classes.button}
                  color="primary"
                  variant="contained"
                  onClick={() => joinBeta()}
                >
                  {t("settings.betaBanner.action")}
                </Button>
              </Grid>
            </Grid>
          </Box>
        </Container>
      </Paper>
      <Divider />
    </>
  );
}

export default Banner;
