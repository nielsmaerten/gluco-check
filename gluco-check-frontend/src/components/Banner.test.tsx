import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { axe } from "jest-axe";
import Banner from "./Banner";

jest.mock("react-i18next", () => ({
  useTranslation: () => {
    return {
      i18n: {
        language: "debug",
      },
      t: jest.fn().mockImplementation((i) => {
        return i;
      }),
    };
  },
}));

describe("Banner", () => {
  const originalLocation = window.location;

  beforeEach(() => {
    // @ts-ignore
    delete window.location;
    window.location = {
      ...originalLocation,
      assign: jest.fn(),
    };
  });

  afterEach(() => {
    window.location = originalLocation;
  });

  it("renders the component", () => {
    const { container } = render(<Banner />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("navigates when button is clicked", async () => {
    render(<Banner />);
    const button = await screen.findByRole("button");
    await userEvent.click(button);
    expect(window.location.assign).toHaveBeenCalled();
  });

  it("has no axe violations", async () => {
    const { container } = render(<Banner />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
