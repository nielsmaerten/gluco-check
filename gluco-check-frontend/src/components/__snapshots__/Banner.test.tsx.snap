// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders the component 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd"
  >
    <div
      class="MuiBox-root MuiBox-root-5"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-flex-start"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-icon-3 MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <p
            class="MuiTypography-root makeStyles-title-1 MuiTypography-body1 MuiTypography-colorSecondary"
          >
            settings.betaBanner.title
          </p>
          <p
            class="MuiTypography-root makeStyles-subtitle-2 MuiTypography-body1"
          >
            settings.betaBanner.message
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-8 MuiGrid-justify-xs-flex-end"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-4 MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              settings.betaBanner.action
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
